packages/fxa-payments-server/server/lib/routes/post-metrics.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const joi = require('joi');
+const amplitude = require('../amplitude');
+
+// TODO: add more validators, and extract validators duplicated here and in
+//       fxa-content-server into fxa-shared
+const clientMetricsConfig = config.get('client_metrics');
+const MAX_EVENT_OFFSET = clientMetricsConfig.maxEventOffset;
+const EVENT_TYPE_PATTERN = /^[\w\s.:-]+$/; // the space is to allow for error contexts that contain spaces, e.g., `error.unknown context.auth.108`
+const BODY_SCHEMA = {
+  data: joi.object().required(),
+  events: joi
+    .array()
+    .items(
+      joi.object().keys({
+        offset: OFFSET_TYPE.max(MAX_EVENT_OFFSET).required(),
+        type: STRING_TYPE.regex(EVENT_TYPE_PATTERN).required(),
+      })
+    )
+    .required(),
+};
+
+module.exports = {
+  method: 'post',
+  path: '/metrics',
+  validate: {
+    body: BODY_SCHEMA,
+  },
+  preProcess: function(req, res, next) {
+    // convert text/plain types to JSON for validation.
+    if (/^text\/plain/.test(req.get('content-type'))) {
+      try {
+        req.body = JSON.parse(req.body);
+      } catch (error) {
+        logger.error(error);
```

Comment:
TODO: not sure if I need to require logger here, or if the logger middleware takes care of it

Reply:
The content-server requires a logger here: https://github.com/mozilla/fxa/blob/56e95b4e68c13399ec988e61f74fa7dcc241ff6f/packages/fxa-content-server/server/lib/routes/post-metrics.js#L11